### PR TITLE
Add cache busting to custom image check + blacklist

### PR DESCRIPTION
**Pull request by indolering**
Fixes [issue](https://github.com/HubSpot/offline/issues/20) wherein custom image checks don't use cache busting, preventing repeat checks from actually checking anything.

### CODE REVIEW

**Comment by indolering** [comment]
Apparently this also sucked up commits related to adding a feature for blacklisting Ajax URLs.  I was planning on making a separate PR with full documentation as well, but it got globbed in with the cache-busting PR.

Before being added to the held queue, a request is tested against every item in the blacklist array, if there is a match it is prevented from being added to the held array.  If the blacklist item is a string, the test is a simple search of the Ajax URL using indexOf.  If the blacklist item is a regex, it performs a regex.test(URL).

File: coffee/offline.coffee [hunk 1]
```diff
@@ -181,7 +186,7 @@ Offline.checks.image = ->
   img = document.createElement 'img'
   img.onerror = Offline.markDown
   img.onload = Offline.markUp
-  img.src = Offline.getOption('checks.image.url')
+  img.src = Offline.getOption('checks.image.url') + cacheBust()
```
Comment (rawls238) [inline coffee/offline.coffee]: what if this url already has query params?

**Comment by rawls238** [comment]
can you please add the blacklist change in a separate PR?

File: coffee/offline.coffee (hunk 1 above)
Comment (indolering) [inline coffee/offline.coffee]: > what if this url already has query params?

Good question, I could add detection of the query parameters and just tack on a non-sense parameter.  Would that be good enough?

**Comment by indolering** [comment]
Yes, there is also a bug in my blacklist implementation.

File: coffee/offline.coffee (hunk 1 above)
Comment (rawls238) [inline coffee/offline.coffee]: yea I think that would work

**Pull request closed without merging**